src/compiler/program.ts

```diff
@@ -690,7 +706,11 @@ namespace ts {
         function getCommonSourceDirectory() {
             if (commonSourceDirectory === undefined) {
                 const emittedFiles = filter(files, file => sourceFileMayBeEmitted(file, options, isSourceFileFromExternalLibrary));
-                if (options.rootDir && checkSourceFilesBelongToPath(emittedFiles, options.rootDir)) {
+                if (options.composite) {
+                    // Project compilations never infer their root from the input source paths
+                    commonSourceDirectory = getNormalizedAbsolutePath(options.rootDir || getDirectoryPath(normalizeSlashes(options.configFilePath)), currentDirectory);
+                }
+                else if (options.rootDir && checkSourceFilesBelongToPath(emittedFiles, options.rootDir)) {

```

Comment:
so what happenes if we set it if `rootDir` is set? we will report the error later on any ways..

the reason i ask, is it will make more sense to write it as:
```ts
if (options.rootDir) {
    commonSourceDirectory = getNormalizedAbsolutePath(options.rootDir, currentDirectory);
}
else if (options.composite) {
    commonSourceDirectory = getDirectoryPath(normalizeSlashes(options.configFilePath)
}
else  {
    commonSourceDirectory = computeCommonSourceDirectory(emittedFiles);
}
```

Reply:
Changed to something similar